parts/k8s/cloud-init/arc-onboarding.yaml

```diff
@@ -0,0 +1,92 @@
+---
```

Comment:
So, as I read through this, I think this is probably O.K. to implement as an addon. I don't think there are any mutations to any of the resources once the onboarding finishes (it's a one-time process).

The only thing to consider is: what would happen if a user manually deletes the job object? According to the official docs, Job resources (and their resultant Pod resources) will stick around forever:

https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-termination-and-cleanup

Is it O.K. if a user deletes the Job object, and then the kube-addon-manager notices and re-runs it? Is the azure-arc-onboarding operation idempotent?

Reply:
Just checked, at this point the onboarding job is idempotent.
I can move to the addon approach.